### ISSUE TITLE
[Warning] Illegal offset type

### ISSUE THREAD

**Issue by spekulatius** [Taitava/silverstripe-slickcarousel#9]
Hello @Taitava 

I've got an upgrade project using your module and found something odd after upgrading from SS3 to SS4. The following error gets thrown on pages using the carousel extension:

![image](https://user-images.githubusercontent.com/8433587/61033292-5f1fb180-a3ed-11e9-821b-7c0f60d99d30.png)

I've followed it back and found the problem in line 39 of CarouselExtension.php:

    $gridfield_config->addComponent(Injector::inst()->create($sortable_rows_class,['Sort']));

The 'sort' is an array while a string is expected to match the key in the error. Not entirely sure why it is an array. Should I submit a PR for this?

Peter

Cheers,
Peter

**Comment by Taitava** [comment]
Hi Peter!I'm currently on a holiday trip which will last for a few weeks. After that I will have a better chance to examine this issue (now I'm writing this message by using my cell phone). Anyway, a pull request sounds good and I will check it after my trip. Thank you for your effort! :)While your pull request will be waiting to be merged, you can use your own forked repository in composer.json to be able to apply the fix in your project. Please let me know if you need further instructions on how to do this.11.7.2019 10.28 Peter Thaleikis <notifications@github.com> kirjoitti:Hello @Taitava
I've got an upgrade project using your module and found something odd after upgrading from SS3 to SS4. The following error gets thrown on pages using the carousel extension:

I've followed it back and found the problem in line 39 of CarouselExtension.php:
$gridfield_config->addComponent(Injector::inst()->create($sortable_rows_class,['Sort']));

The 'sort' is an array while a string is expected to match the key in the error. Not entirely sure why it is an array. Should I submit a PR for this?
Peter
Cheers,
Peter

**Comment by spekulatius** [comment]
Hello Jarkko,

no problem at all. Enjoy your holiday. I'll prepare a PR over the next days
and send it to you - no rush on checking it out. I'll add my fork as a
dependency for now :)

Peter

**Comment by spekulatius** [comment]
Hello Jarkko,

I've opened the PR for you. No need to check it now, it can wait until you
are back.

Peter

**Comment by Taitava** [comment]
Hi Peter,

sorry that this took so long, but I have now checked this.

> The 'sort' is an array while a string is expected to match the key in the error. Not entirely sure why it is an array.

It seems that this was a mistake that I made when upgrading this module to SS 4. While I did the upgrade, I changed the way how an instance of `UndefinedOffset\SortableGridField\Forms\GridFieldSortableRows` class is created to use SilverStripe's `Injector` class to create the instance (just to make instance creation to comply with the standard SilverStripe way) (commit 7ea55b3a4637b28ca92e428b2aed0733b329a320). While doing this change, I somehow managed to screw up the 'Sort' parameter. Perhaps I thought that the `create()` method of the `Injector` class would want to take pass through parameters as an array, but for some reason I didn't check it.

So your solution seems to be correct. Thank you for your effort! :)
